src/VisualStudio/LiveShare/Impl/ClassificationsHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.CustomProtocol;
+using Microsoft.VisualStudio.LanguageServices.LiveShare.CustomProtocol;
+using Microsoft.VisualStudio.LiveShare.LanguageServices;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare
+{
+    /// <summary>
+    /// Handler for a request to classify the document. This is used for semantic colorization and only works for C#\VB.
+    /// TODO - Move once defined as a custom protocol.
+    /// </summary>
+    [ExportLspRequestHandler(LiveShareConstants.RoslynContractName, RoslynMethods.ClassificationsName)]
+    internal class ClassificationsHandler : ILspRequestHandler<ClassificationParams, ClassificationSpan[], Solution>
```

Comment:
we need to talk to Sri so that we can have LspRequestHandler for each language rather than like RoslynContactName. I think we want to be explicit about what language this actually support since some of this support F# and typescript as well?

Reply:
https://github.com/dotnet/roslyn/projects/45#card-20956018